app/register/RegisterForm.tsx

```diff
@@ -0,0 +1,83 @@
+"use client";
+
+import FormInput from "@/components/FormInput";
+import styles from "../page.module.css";
+import { SyntheticEvent } from "react";
+
+const handleSubmit = async (e: SyntheticEvent) => {
+  e.preventDefault();
+  const target = e.target as typeof e.target & {
+    firstName: { value: string };
+    lastName: { value: string };
+    email: { value: string };
+    password: { value: string };
+    role: { value: string };
+  };
+
+  const body = JSON.stringify({
+    firstName: target.firstName.value,
+    lastName: target.lastName.value,
+    email: target.email.value,
+    password: target.password.value,
+    role: 'Admin',
+  });
+
+  const res = await fetch("http://localhost:8080/auth/register", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: body,
+  });
+  const data = await res.json();
```

Comment:
Same thing here; use Axios & don't hard-code the URL.

Reply:
Listo